### ISSUE TITLE
Improve logging to debug issues faster

### ISSUE THREAD

**Issue by abhayana24** [cisco/mindmeld#232]
I started with understanding mindmeld and built a simple bot on my custom data. In my `mapping.json` file for one of the entities didn't had `entities` key as I missed it while creating the mapping.json file, so although the json format was correct, but didn't conform to mindmeld's way of creating the `mapping.json` file. When I build the models, I got error because of this issue, but the problem was the logging/error messages was not at all helpful in debuging the cause of the issue. It would be great if we can improve error logging, it will make life of someone just starting with mindmeld easier. I am attaching the screenshots of the error.

![image](https://user-images.githubusercontent.com/7872601/93773375-af271480-fc3d-11ea-8e5b-d3372a1351fd.png)
![image](https://user-images.githubusercontent.com/7872601/93773443-c5cd6b80-fc3d-11ea-8b5b-71e43a34ae1b.png)

**Comment by minhtuev** [comment]
Got it, have you been able to build the MM app otherwise?

**Comment by abhayana24** [comment]
Yes @minhtuev after figuring out the issue(which took quite some time) I was able to build the app.

**Comment by minhtuev** [comment]
Got it, we will take a look and add some friendly error messages :)

**Comment by ProsperousHeart** [comment]
In the mean time, [here](https://www.kdnuggets.com/2021/12/3-tools-track-visualize-execution-python-code.html) are some nifty packages you can add to your own local machine to get a better idea of when things fail.
